tree/treeplayer/src/TCsvDS.cxx

```diff
@@ -277,17 +276,14 @@ TCsvDS::TCsvDS(std::string_view fileName, bool readHeaders, char delimiter) // T
       // Infer types of columns with first record
       InferColTypes(columns);
 
-      // Read all records and store them in memory
-      do {
-         fRecords.emplace_back();
-         FillRecord(line, fRecords.back());
-      } while (std::getline(stream, line));
+      // Fill with the content of the first line
```

Comment:
why do we need to read a line of the body here?
if we did it afterwards we would not need to have ifs on `fEntryRangesRequested` in `GetEntryRanges`

Reply:
We need to read a line in the constructor to infer the types from the content of that line.

Reply:
ah I see!